agent/main.go

```diff
@@ -94,17 +94,17 @@ func main() {
 			panic("API server address must be provided with --api-server-address when using --tap")
 		}
 
+		hostMode := os.Getenv(shared.HostModeEnvVar) == "1"
+		tapOpts := &tap.TapOpts{HostMode: hostMode}
 		tapTargets := getTapTargets()
 		if tapTargets != nil {
-			tap.SetFilterAuthorities(tapTargets)
-			logger.Log.Infof("Filtering for the following authorities: %v", tap.GetFilterIPs())
+			tapOpts.FilterAuthorities = tapTargets
```

Comment:
From the PR description: @nimrod-up9 
The PR force filter authorities to be passed only once instead of multiple times, It does it by moving tap.SetFilterAuthorities to TapOpts parameter of StartPassiveTapper. The motivation for it is because the behavior of the tapper changed, adding and removing Authorities requires extra work. At this point we don't need this ability, hence we prevent it by removing tap.SetFilterAuthorities.

---

Regarding the movement of the condition (the second comment), the tapper package should protect itself from nil (panic). It can't count on the caller to set the right defaults for it, that's why I put the protection in the entrypoint to tapper (StartPassiveTapper)